### PR TITLE
OpInfo: true_divide and minor fix

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #54261

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -6695,15 +6699,6 @@ def method_tests():
...
         ('div', uniform_scalar(1e-1, requires_grad=True), (3.14,), 'scalar_constant', (True,)),
-        ('true_divide', (S, S, S), (torch.rand(S, S, S) + 0.1,), '', (True,)),
-        ('true_divide', (S, S, S), (torch.rand(S, S) + 0.1,), 'broadcast_rhs', (True,)),
-        ('true_divide', (S, S), (torch.rand(S, S, S) + 0.1,), 'broadcast_lhs', (True,)),
-        ('true_divide', (S, 1, S), (torch.rand(M, S) + 0.1,), 'broadcast_all', (True,)),
-        ('true_divide', (), (uniform_scalar(0.1),), 'scalar', (True,)),
-        ('true_divide', (S, S, S), (uniform_scalar(0.1),), 'scalar_broadcast_rhs', (True,)),
-        ('true_divide', (), (uniform_scalar(0.1),), 'scalar_broadcast_lhs', (True,)),
-        ('true_divide', torch.rand(S, S, S) + 1e-1, (3.14,), 'constant', (True,)),
-        ('true_divide', uniform_scalar(1e-1, requires_grad=True), (3.14,), 'scalar_constant', (True,)),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:6706]: Cases covered by binary_pwise,

https://github.com/pytorch/pytorch/blob/41054f2ab5bb39d28a3eb8497f1a65b42385a996/torch/testing/_internal/common_methods_invocations.py#L727-L741

**Comment by mruberry** [comment]
Another NNC issue:

```
======================================================================
ERROR [0.366s]: test_unsupported_true_divide_cpu_float32 (__main__.TestNNCOpInfoCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 297, in instantiated_test
    raise rte
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 292, in instantiated_test
    result = test_fn(self, *args)
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 266, in test_wrapper
    return test(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 729, in only_fn
    return fn(slf, device, *args, **kwargs)
  File "test_jit_fuser_te.py", line 2001, in test_unsupported
    raise RuntimeError("Expected test to fail. If it now works, move op into works_list")
RuntimeError: Expected test to fail. If it now works, move op into works_list
```

cc @Chillee 

Also we should rename that test to be more specific, like "test_nnc_unsupported"

**Comment by mruberry** [comment]
> Another NNC issue:
> 
> ```
> ======================================================================
> ERROR [0.366s]: test_unsupported_true_divide_cpu_float32 (__main__.TestNNCOpInfoCPU)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 297, in instantiated_test
>     raise rte
>   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 292, in instantiated_test
>     result = test_fn(self, *args)
>   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 266, in test_wrapper
>     return test(*args, **kwargs)
>   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 729, in only_fn
>     return fn(slf, device, *args, **kwargs)
>   File "test_jit_fuser_te.py", line 2001, in test_unsupported
>     raise RuntimeError("Expected test to fail. If it now works, move op into works_list")
> RuntimeError: Expected test to fail. If it now works, move op into works_list
> ```
> 
> cc @Chillee
> 
> Also we should rename that test to be more specific, like "test_nnc_unsupported"

And can we move it into test_ops.py? That way people only have to run one test file when adding an OpInfo. The jit support tests are also in test_ops.py, for example.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -744,8 +744,9 @@ def sample_inputs_binary_pwise(op_info, device, dtype, requires_grad, **kwargs):
     for first_shape, shape_or_scalar, broadcasts_input in test_cases:
         arg = shape_or_scalar
         if isinstance(shape_or_scalar, tuple):
+            exclude_zero = kwargs.get('rhs_exclude_zero', False)
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:747]: Let's extend the comment documentation with what kwargs are currently supported and what they do, too

File: torch/testing/_internal/common_utils.py [hunk 3]
```diff
@@ -1789,9 +1789,8 @@ def make_tensor(size, device: torch.device, dtype: torch.dtype, *, low=None, hig
         else:
             assert dtype in complex_types()
             float_dtype = torch.float if dtype is torch.cfloat else torch.double
-            real = torch.tensor(torch.finfo(float_dtype).eps, device=device, dtype=dtype)
-            imag = torch.tensor(torch.finfo(float_dtype).eps, device=device, dtype=dtype)
-            replace_with = torch.complex(real, imag)
+            float_eps = torch.tensor(torch.finfo(float_dtype).eps, device=device, dtype=float_dtype)
```
Comment (mruberry) [inline torch/testing/_internal/common_utils.py:1792]: Cool

**Review by mruberry — APPROVED** [review]
Nice work @kshitij12345. This needs a fix for the NNC test and a small comment update.

I also left a couple notes for @Chillee to move that test into test_ops.py (if possible), improve its naming, and (these notes are elsewhere) add metadata to OpInfos for whether they support NNC or not. That should make it easier to work with that test. But we should do any of those things in this PR.

Is div next?

**Comment by Chillee** [comment]
> I also left a couple notes for @Chillee to move that test into test_ops.py (if possible), improve its naming, and (these notes are elsewhere) add metadata to OpInfos for whether they support NNC or not. That should make it easier to work with that test. But we should do any of those things in this PR.

These changes sound fine to me - the main consideration is just maximizing the ease of adding new OpInfos + maintainibility of these OpInfos.

I'd be glad to put up a PR for it, although I'm PTO the next 2 weeks, so it's possible I won't get to it (unless I do it now... maybe). Feel free to move the code to where you'd like.

**Review by krshrimali — APPROVED** [review]
LGTM, @kshitij12345!

**Comment by mruberry** [comment]
> > I also left a couple notes for @Chillee to move that test into test_ops.py (if possible), improve its naming, and (these notes are elsewhere) add metadata to OpInfos for whether they support NNC or not. That should make it easier to work with that test. But we should do any of those things in this PR.
> 
> These changes sound fine to me - the main consideration is just maximizing the ease of adding new OpInfos + maintainibility of these OpInfos.
> 
> I'd be glad to put up a PR for it, although I'm PTO the next 2 weeks, so it's possible I won't get to it (unless I do it now... maybe). Feel free to move the code to where you'd like.

Enjoy your PTO! I filed https://github.com/pytorch/pytorch/issues/59185#issue-906673386 to track this

**Pull request closed without merging**